File: src/main/scala/stni/atlassian/remote/query/QueryIssueWrapper.scala
package stni.atlassian.remote.query

import com.atlassian.jira.rpc.soap.beans.{RemoteComponent, RemoteCustomFieldValue, RemoteVersion}
import java.util.Calendar

/**
 *
 */
abstract class QueryIssueWrapper(elems: Seq[QueryIssue], t: QueryIssue => QueryIssueList) extends AbstractQueryList[QueryIssue, QueryIssueList](elems, t) {
  def issue = elems(0).issue

  def getAffectsVersions: Array[RemoteVersion] = issue.getAffectsVersions

  def getCreated: Calendar = issue.getCreated

  def getStatus: String = issue.getStatus

  def getDuedate: Calendar = issue.getDuedate

  def getCustomFieldValues: Array[RemoteCustomFieldValue] = issue.getCustomFieldValues

  def getComponents: Array[RemoteComponent] = issue.getComponents

  def getEnvironment: String = issue.getEnvironment

  def getType: String = issue.getType

  def getAssignee: String = issue.getAssignee

  def getResolution: String = issue.getResolution

  def getPriority: String = issue.getPriority

  def getAttachmentNames: Array[String] = issue.getAttachmentNames

  def getFixVersions: Array[RemoteVersion] = issue.getFixVersions

  def getUpdated: Calendar = issue.getUpdated

  def getReporter: String = issue.getReporter

  def getId: String = issue.getId

  def getSummary: String = issue.getSummary

  def getVotes: Long = issue.getVotes

  def getProject: String = issue.getProject

  def getDescription: String = issue.getDescription

  def getKey: String = issue.getKey

  override def toString = "Issue (key=" + getKey + ")"
}

File: src/main/java/stni/atlassian/remote/queryold/QueryIssueWrapper.java
package stni.atlassian.remote.queryold;

import com.atlassian.jira.rpc.soap.beans.RemoteComponent;
import com.atlassian.jira.rpc.soap.beans.RemoteCustomFieldValue;
import com.atlassian.jira.rpc.soap.beans.RemoteIssue;
import com.atlassian.jira.rpc.soap.beans.RemoteVersion;

import java.util.Calendar;

/**
 *
 */
abstract class QueryIssueWrapper extends AbstractQueryObject<QueryIssue> {
    private final RemoteIssue issue;

    QueryIssueWrapper(JiraQuery query, RemoteIssue issue) {
        super(query);
        this.issue = issue;
    }

    public RemoteIssue getIssue() {
        return issue;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }

        QueryIssueWrapper that = (QueryIssueWrapper) o;
        if (!issue.equals(that.issue)) {
            return false;
        }
        return true;
    }

    @Override
    public int hashCode() {
        int result = super.hashCode();
        result = 31 * result + issue.hashCode();
        return result;
    }

    @Override
    public String toString() {
        return "QueryIssueWrapper{" +
                "issue=" + issue +
                '}';
    }

    public RemoteVersion[] getAffectsVersions() {
        return issue.getAffectsVersions();
    }

    public Calendar getCreated() {
        return issue.getCreated();
    }

    public String getStatus() {
        return issue.getStatus();
    }

    public Calendar getDuedate() {
        return issue.getDuedate();
    }

    public RemoteCustomFieldValue[] getCustomFieldValues() {
        return issue.getCustomFieldValues();
    }

    public RemoteComponent[] getComponents() {
        return issue.getComponents();
    }

    public String getEnvironment() {
        return issue.getEnvironment();
    }

    public String getType() {
        return issue.getType();
    }

    public String getAssignee() {
        return issue.getAssignee();
    }

    public String getResolution() {
        return issue.getResolution();
    }

    public String getPriority() {
        return issue.getPriority();
    }

    public String[] getAttachmentNames() {
        return issue.getAttachmentNames();
    }

    public RemoteVersion[] getFixVersions() {
        return issue.getFixVersions();
    }

    public Calendar getUpdated() {
        return issue.getUpdated();
    }

    public String getReporter() {
        return issue.getReporter();
    }

    public String getId() {
        return issue.getId();
    }

    public String getSummary() {
        return issue.getSummary();
    }

    public Long getVotes() {
        return issue.getVotes();
    }

    public String getProject() {
        return issue.getProject();
    }

    public String getDescription() {
        return issue.getDescription();
    }

    public String getKey() {
        return issue.getKey();
    }
}